src/Worker/GrpcInvoker.php

```diff
@@ -79,16 +79,30 @@ public function invokeFinalize(): void
         }
 
         if ($this->kernel instanceof RebootableInterface && $this->dependencies->getKernelRebootStrategy()->shouldReboot()) {
+            try {
+                if ($this->kernel->getContainer()->has('services_resetter')) {
+                    /** @var ResetInterface $resetter */
+                    $resetter = $this->kernel->getContainer()->get('services_resetter');
+                    $resetter->reset();
+                }
+            } catch (\Throwable $e) {
+                $this->logger->error(
+                    \sprintf(
+                        'An error occurred when resetting services: %s',
+                        $e->getMessage()
+                    ),
+                    ['exception' => $e]
+                );
+
+                throw $e;
+            }
+
             $this->kernel->reboot(null);
             /** @var GrpcDependencies */
             $deps = $this->kernel->getContainer()->get(GrpcDependencies::class);
 
             $this->dependencies = $deps;
             $this->dependencies->getEventDispatcher()->dispatch(new WorkerKernelRebootedEvent());
-        } elseif ($this->kernel->getContainer()->has('services_resetter')) {
-            /** @var ResetInterface $resetter */
-            $resetter = $this->kernel->getContainer()->get('services_resetter');
-            $resetter->reset();
```

Comment:
We need to keep this for non-rebootable Kernels and for when we should not reboot.

Reply:
brought it back , add to top

Reply:
sounds reasonable